### PR TITLE
add support for opencloudos / tencentos

### PR DESCRIPTION
**Pull request by wswsmao**
There is so many developers are try to install openvino in opencloudos and tencentos.
This pr add support for opencloudos / tencentos.

OpenCloudOS is an open source operating system community jointly initiated by Tencent and its partners.

for more news:
https://www.opencloudos.org/
https://docs.opencloudos.org/en/

### CODE REVIEW

File: scripts/install_dependencies/install_openvino_dependencies.sh [hunk 1]
```diff
@@ -96,6 +96,9 @@ if [ "$os" == "auto" ] ; then
     case $os in
         centos7|centos8|centos9|\
         rhel8|rhel9.1|\
+        opencloudos8.5|opencloudos8.6|opencloudos8.8|opencloudos9.0|opencloudos9.2|\
+        opencloudos2301|opencloudos2305|opencloudos2309|opencloudos2401|opencloudos2403|\
+        tencentos3.1|tencentos3.2|tencentos3.3|tencentos4.0|tencentos4.2|\
```
Comment (ilya-lavrenov) [inline scripts/install_dependencies/install_openvino_dependencies.sh:101]: what docker images can I use to test it?

Comment (wswsmao) [reply scripts/install_dependencies/install_openvino_dependencies.sh:101]: opencloudos：
https://hub.docker.com/r/opencloudos/opencloudos/tags

Comment (wswsmao) [reply scripts/install_dependencies/install_openvino_dependencies.sh:101]: tencentos
https://hub.docker.com/r/tencentos/tencentos4-minimal

Comment (wswsmao) [reply scripts/install_dependencies/install_openvino_dependencies.sh:101]: > is it possible to add support to https://github.com/openvinotoolkit/openvino/blob/master/install_build_dependencies.sh as well? I've tried to build inside opencloudos docker image and that script have not installed build dependencies.

ok，I can add it.
But I notice that the conditions in master/install_build_dependencies.sh are different with scripts/install_dependencies/install_openvino_dependencies.sh, for example, there is not anolis or openEuler20.03 branch in install_build_dependencies.sh.
How should add support to install_build_dependencies.sh?

Comment (ilya-lavrenov) [reply scripts/install_dependencies/install_openvino_dependencies.sh:101]: `install_build_dependencies.sh` has more generic conditions, which check some files to detect OS.
`install_openvino_dependencies.sh` has conditions on specific OS name and version.

In future we want to unify both scripts to have a common conditions. But currently, you need to have a generic condition like https://github.com/openvinotoolkit/openvino/blob/master/install_build_dependencies.sh#L92 to detect a set of OSes

Comment (wswsmao) [reply scripts/install_dependencies/install_openvino_dependencies.sh:100]: > `install_build_dependencies.sh` has more generic conditions, which check some files to detect OS. `install_openvino_dependencies.sh` has conditions on specific OS name and version.
> 
> In future we want to unify both scripts to have a common conditions. But currently, you need to have a generic condition like https://github.com/openvinotoolkit/openvino/blob/master/install_build_dependencies.sh#L92 to detect a set of OSes

done

File: install_build_dependencies.sh [hunk 2]
```diff
@@ -294,6 +294,44 @@ elif [ -f /etc/os-release ] && grep -q "alpine" /etc/os-release; then
         enchant2 \
         `# samples` \
         nlohmann-json
+elif [ -f /etc/os-release ] && \
+   (grep -q "tencentos" /etc/os-release || grep -q "opencloudos" /etc/os-release); then
```
Comment (ilya-lavrenov) [inline install_build_dependencies.sh:298]: can it be merged with `elif [ -f /etc/redhat-release ] || grep -q "rhel" /etc/os-release ; then` branch?
Looks like both OSes use the same package manager

Comment (wswsmao) [reply install_build_dependencies.sh]: > can it be merged with `elif [ -f /etc/redhat-release ] || grep -q "rhel" /etc/os-release ; then` branch? Looks like both OSes use the same package manager

done

File: scripts/install_dependencies/install_openvino_dependencies.sh (hunk 1 above)
Comment (wswsmao) [reply scripts/install_dependencies/install_openvino_dependencies.sh:100]: > I've found that `patchelf` and `ShellCheck` are not available on these docker images out of box. Maybe different names should be used? or some extra repositories should be enabled
> 
> In worst case, we can add conditions to skip these dependencies for these OSes

OK，These packages are supported last month, and synchronizing to the repo may take some time
https://gitee.com/opencloudos-stream/ShellCheck
https://gitee.com/opencloudos-stream/patchelf

I will add conditions temporarily

Comment (wswsmao) [reply scripts/install_dependencies/install_openvino_dependencies.sh:100]: > I've found that `patchelf` and `ShellCheck` are not available on these docker images out of box. Maybe different names should be used? or some extra repositories should be enabled
> 
> In worst case, we can add conditions to skip these dependencies for these OSes

done

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@wswsmao thank you for contribution!
Last ask is to update this file https://github.com/openvinotoolkit/openvino/blob/master/.github/dockerfiles/docker_tag with your PR number. It will be used to tag docker image with infrastructure. 

Also, pay attention to `PR Commits / Checks` check. Looks like you need proper identification on github.

**Comment by wswsmao** [comment]
> @wswsmao thank you for contribution! Last ask is to update this file https://github.com/openvinotoolkit/openvino/blob/master/.github/dockerfiles/docker_tag with your PR number. It will be used to tag docker image with infrastructure.
> 
> Also, pay attention to `PR Commits / Checks` check. Looks like you need proper identification on github.

should I update docker_tag  by this pr or create a new pr?

**Comment by wswsmao** [comment]
> @wswsmao thank you for contribution! Last ask is to update this file https://github.com/openvinotoolkit/openvino/blob/master/.github/dockerfiles/docker_tag with your PR number. It will be used to tag docker image with infrastructure.
> 
> Also, pay attention to `PR Commits / Checks` check. Looks like you need proper identification on github.

about Commits / Checks:
```
    ERROR: User with the commit author email is absent in GitHub: Shuo Wang
    ERROR: User with the commit committer email is absent in GitHub: Shuo Wang
```

is it necessary to change committer email to abushwang ?

**Comment by wswsmao** [comment]
> @wswsmao thank you for contribution! Last ask is to update this file https://github.com/openvinotoolkit/openvino/blob/master/.github/dockerfiles/docker_tag with your PR number. It will be used to tag docker image with infrastructure.
> 
> Also, pay attention to `PR Commits / Checks` check. Looks like you need proper identification on github.

all right,  I have add pr num and update my identification

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by wswsmao** [comment]
> > @wswsmao thank you for contribution! Last ask is to update this file https://github.com/openvinotoolkit/openvino/blob/master/.github/dockerfiles/docker_tag with your PR number. It will be used to tag docker image with infrastructure.
> > Also, pay attention to `PR Commits / Checks` check. Looks like you need proper identification on github.
> 
> all right, I have add pr num and update my identification

@ilya-lavrenov the checks seems like finally passed

File: .github/dockerfiles/docker_tag [hunk 3]
```diff
@@ -1 +1,2 @@
-pr-24742
\ No newline at end of file
+pr-24742
+pr-24878
```
[the remark this answers is not in the document]
Comment (wswsmao) [reply .github/dockerfiles/docker_tag:1]: > you need to replace old with new one

done

**Comment by ilya-lavrenov** [comment]
build_jenkins

**Pull request merged**